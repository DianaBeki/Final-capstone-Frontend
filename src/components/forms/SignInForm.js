import React from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { signInUser } from '../../redux/user/userSlice';

const SignInForm = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const handleSignIn = async (e) => {
    e.preventDefault();

    try {
      await dispatch(signInUser({ name: email }));
      navigate.push('/doctors');
    } catch (error) {
      console.error('Sign In Error:', error);
    }
  };
  eturn(
    <div className="flex items-center justify-center h-screen">
      <form className="bg-white p-8 shadow-md rounded-md w-96" onSubmit={handleSignIn}>
        <h2 className="text-2xl font-semibold mb-6">Sign In</h2>
        <div className="mb-4">
          <input
            type="email"
            id="email"
            name="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            className="mt-1 p-2 w-full border rounded-md"
            placeholder="Enter your email"
          />
        </div>
        <div className="mb-4">
          <input
            type="password"
            id="password"
            name="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            className="mt-1 p-2 w-full border rounded-md"
            placeholder="Enter your password"
          />
        </div>
        <button
          type="submit"
          className="bg-blue-500 text-white px-4 py-2 rounded-md hover:bg-blue-600"
        >
          Sign In
        </button>
      </form>
    </div>,
  );
};

export default SignInForm;
