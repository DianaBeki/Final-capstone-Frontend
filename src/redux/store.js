import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './doctor/doctorSlice';
import userSlice from './user/userSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
    user: userSlice,
  },
});

export default store;
