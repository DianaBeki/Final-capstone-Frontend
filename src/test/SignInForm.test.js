import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import SignInForm from '../components/forms/SignInForm';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    <Router>
      {component}
    </Router>
  </Provider>,
);

describe('SignInForm Component', () => {
  it('renders SignInForm component', () => {
    const { container } = render(<SignInForm />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
