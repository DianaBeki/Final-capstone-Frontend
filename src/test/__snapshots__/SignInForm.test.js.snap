// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm Component renders SignInForm component 1`] = `
<div
  class="flex items-center justify-center h-screen"
>
  <form
    class="bg-white p-8 shadow-md rounded-md w-96"
  >
    <h2
      class="text-2xl font-semibold mb-6"
    >
      Sign In
    </h2>
    <div
      class="mb-4"
    >
      <input
        class="mt-1 p-2 w-full border rounded-md"
        id="name"
        name="name"
        placeholder="Enter your email"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-4"
    >
      <input
        class="mt-1 p-2 w-full border rounded-md"
        id="password"
        name="password"
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <button
      class="bg-blue-500 text-white px-4 py-2 rounded-md hover:bg-blue-600"
      type="submit"
    >
      Sign In
    </button>
    <p
      class="mt-2"
    >
      Don't have an account?
       
      <a
        href="/"
      >
        Click here to sign_up
      </a>
    </p>
  </form>
</div>
`;
